Janky is acting all weird - entirely unrelated to this PR:

```
00:00:21.192 Step 102/102 : COPY . /go/src/github.com/docker/docker
00:00:21.544  ---> Using cache
00:00:21.544  ---> 406577b37485
00:00:21.661 Successfully built 406577b37485
00:00:21.689 Successfully tagged docker:659ab7f
00:00:22.803 Congratulations!  All commits are properly signed with the DCO!
00:02:23.225 daemon/restart.go:41:64:warning: unnecessary conversion (unconvert)
00:02:23.825 Build step 'Execute shell' marked build as failure
00:02:23.825 Set build name.
00:02:23.826 New build name is '#38782'
00:02:23.854 [PostBuildScript] - Executing post build scripts.
00:02:23.868 [Docker-PRs] $ /bin/sh -xe /tmp/jenkins8763975022947107079.sh
00:02:23.872 + echo Ensuring container killed.
00:02:23.872 Ensuring container killed.
00:02:23.872 + docker rm -vf docker-pr53054
00:02:23.945 Error: No such container: docker-pr53054
00:02:23.946 + true
00:02:23.946 + echo Chowning /workspace to jenkins user
00:02:23.946 Chowning /workspace to jenkins user
00:02:23.946 + id -u
00:02:23.951 + id -g
00:02:23.952 + docker run --rm -v /home/ubuntu/workspace/Docker-PRs:/workspace busybox chown -R 1000:1000 /workspace
00:02:25.064 [Docker-PRs] $ /bin/sh -xe /tmp/jenkins2663173096780292328.sh
00:02:25.073 + echo Creating bundles.tar.gz
00:02:25.073 Creating bundles.tar.gz
00:02:25.074 + xargs tar -czf bundles.tar.gz
00:02:25.074 + find bundles -name *.log -o -name *.prof -o -name integration.test
00:02:25.077 tar: Cowardly refusing to create an empty archive
00:02:25.077 Try 'tar --help' or 'tar --usage' for more information.
00:02:25.077 + true
00:02:25.086 Archiving artifacts
00:02:25.089 Notifying endpoint with url 'https://leeroy.dockerproject.org/notification/jenkins'
00:02:25.801 [WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
00:02:26.212 Finished: FAILURE
```

--- a/daemon/restart.go
+++ b/daemon/restart.go
@@ -3,6 +3,7 @@ package daemon // import "github.com/docker/docker/daemon"
 import (
 	"fmt"
 
+	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
 	"github.com/sirupsen/logrus"
 )
@@ -34,11 +35,23 @@ func (daemon *Daemon) ContainerRestart(name string, seconds *int) error {
 // gracefully stop, before forcefully terminating the container. If
 // given a negative duration, wait forever for a graceful stop.
 func (daemon *Daemon) containerRestart(container *container.Container, seconds int) error {
+
+	// Determine isolation. If not specified in the hostconfig, use daemon default.
+	actualIsolation := container.HostConfig.Isolation
+	if containertypes.Isolation.IsDefault(actualIsolation) {
+		actualIsolation = daemon.defaultIsolation
+	}
+
 	// Avoid unnecessarily unmounting and then directly mounting
 	// the container when the container stops and then starts
-	// again
-	if err := daemon.Mount(container); err == nil {
-		defer daemon.Unmount(container)
+	// again. We do not do this for Hyper-V isolated containers
+	// (implying also on Windows) as the HCS must have exclusive
+	// access to mount the containers filesystem inside the utility
+	// VM.
+	if !containertypes.Isolation.IsHyperV(actualIsolation) {
+		if err := daemon.Mount(container); err == nil {
+			defer daemon.Unmount(container)
+		}
 	}
 
 	if container.IsRunning() {

